pkg/cluster/nodes/nodes.go

```diff
@@ -99,3 +100,35 @@ func list(visit func(string, *Node), filters ...string) error {
 	}
 	return nil
 }
+
+// WaitForReady uses kubectl inside the "node" container to check if the
+// control plane nodes are "Ready".
+func WaitForReady(node *Node, until time.Time) bool {
+	return tryUntil(until, func() bool {
+		cmd := node.Command(
+			"kubectl",
+			"--kubeconfig=/etc/kubernetes/admin.conf",
+			"get",
+			"nodes",
+			"--selector=node-role.kubernetes.io/master",
+			// When the node reaches status ready, the status field will be set
+			// to true.
+			"-o=jsonpath='{.items..status.conditions[-1:].status}'",
+		)
+		lines, err := exec.CombinedOutputLines(cmd)
+		if err != nil {
+			return false
+		}
+		if len(lines) != 1 {
```

Comment:
this parsing will not work for multi-node, we should add a TODO around this at least as we have ongoing work for multi-node now.

Reply:
Completely missed this! Thank you. I think I fixed this